### DISCUSSION TITLE
ros2,ros1 bridge

### DISCUSSION THREAD

**Discussion by markostamos** [ctu-mrs/mrs_uav_system#128] [Q&A]
I've been trying to use this package in ros2  with ros1_bridge.  I built a ros2 workspace with the same custom messages  package(mrs_msg) but when i run the ros1_bridge i can only see the services that don't have custom messages.

Has anyone done anything similar and can help me solve this issue? 

ros1:Noetic
ros2:Galactic

**Comment by klaxalk** [comment]
Hey, I have not tried ROS2 with our system since last year's December. IIRC, the ROS bridge you install from APT can only bridge standard messages. If you want to bridge custom messages, you have to compile the bridge from the source. We have a set of scripts for that [here](https://github.com/ctu-mrs/uav_core/tree/master/installation/ros2). However, this requires complementary versions of mrs_msgs in ROS1 and ROS2. We have a branch of mrs_msgs for ROS2, but nobody used it for a year, so it is not up-to-date. I will check if it can be updated.
